src/main/java/com/aws/greengrass/mqttclient/spool/CloudMessageSpool.java

```diff
@@ -5,11 +5,15 @@
 
 package com.aws.greengrass.mqttclient.spool;
 
+import java.util.List;
+
 public interface CloudMessageSpool {
 
     SpoolMessage getMessageById(long id);
 
     void removeMessageById(long id);
 
     void add(long id, SpoolMessage message);
-}
+
+    List<Long> getAllSpoolMessageIds();
```

Comment:
do we really need a `List` specifically? Can we get away with a generic `Collection`?

Reply:
I think it is better to use a List because the collection needs to retain order. It is mainly used to sync the existing disk queue with the runtime queue. 

Unless Collection is able to retain the order of the messageIds that we get directly from SQLite. 

What do you think?

Reply:
Could use `Iterable` instead, which would be ordered.